packages/pgsearch/client.js

```diff
@@ -237,12 +237,14 @@ class Batch {
 
   async deleteDocument(context) {
     let { branch, type, id } = context;
+    let { rows } = await this.client.query('select branch, type, id, source, generation, upstream_doc as "upstreamDoc" from documents where branch=$1 and type=$2 and id=$3', [branch, type, id]);
+    let [ eventContext={} ] = rows;
 
     this._touched[`${branch}/${type}/${id}`] = this._touchCounter++;
     let sql = 'delete from documents where branch=$1 and type=$2 and id=$3';
 
     await this.client.query(sql, [branch, type, id]);
-    await this.client.emitEvent('delete', { type, id });
+    await this.client.emitEvent('delete', eventContext);
```

Comment:
I would prefer not to leak all the internal implementation details of the rows here. Let's enumerate only the parts that are needed.

Reply:
ok. in this case i need to read the document to see what addresses are being deleted, so i'll just add that to the event.